core/src/test/scala/org/apache/spark/sql/test/SharedSQLContext.scala

```diff
@@ -274,18 +292,32 @@ object SharedSQLContext extends Logging {
       runTPCDS = tpcdsDBName != ""
 
       _tidbConf = prop
+
+      if (isTidbConfigPropertiesInjectedToSparkEnabled) {
+        sparkConf.set(PD_ADDRESSES, pdAddresses)
+        sparkConf.set(ENABLE_AUTO_LOAD_STATISTICS, "true")
+        sparkConf.set("spark.sql.decimalOperations.allowPrecisionLoss", "false")
+        sparkConf.set(REQUEST_ISOLATION_LEVEL, SNAPSHOT_ISOLATION_LEVEL)
+        sparkConf.set("spark.sql.extensions", "org.apache.spark.sql.TiExtensions")
+        sparkConf.set(DB_PREFIX, dbPrefix)
+      }
+
       _sparkSession = new TestSparkSession(sparkConf, isHiveEnabled).session
     }
 
   /**
    * Make sure the [[TestSparkSession]] is initialized before any tests are run.
    */
-  def init(forceNotLoad: Boolean = false, isHiveEnabled: Boolean = false): Unit = {
-    initializeConf(isHiveEnabled)
+  def init(forceNotLoad: Boolean = false,
+           isHiveEnabled: Boolean = false,
+           isTidbConfigPropertiesInjectedToSparkEnabled: Boolean = true): Unit = {
+    initializeConf(isHiveEnabled, isTidbConfigPropertiesInjectedToSparkEnabled)
     initializeSparkSession()
     initializeTiDB(forceNotLoad)
     initializeJDBC()
-    initializeTiContext()
+    if (isTidbConfigPropertiesInjectedToSparkEnabled) {
```

Comment:
is it true that in such cases, we do not need TiContext?

Reply:
if isTidbConfigPropertiesInjectedToSparkEnabled==false, user do not need to create `TiContext`. `TiContext` will be created by `com.pingcap.tispark.DefaultSource`.